### PR TITLE
Fix message for snapshots with evaluation errors

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
set the message in snapshot with the content of the first evaluation errors

# Motivation

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [DEBUG-2704]

[DEBUG-2704]: https://datadoghq.atlassian.net/browse/DEBUG-2704?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java [hunk 1]
```diff
@@ -572,24 +572,30 @@ protected boolean fillSnapshot(
...
       shouldCommit = true;
     }
     return shouldCommit;
   }
 
+  private static void fillEvaluationErrors(
+      CapturedContext context, Snapshot snapshot, LogStatus status) {
+    if (context.getCapturedThrowable() != null) {
+      // report also uncaught exception
+      snapshot.setEntry(context);
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java:589]: we always call snapshot.setEntry here... I think this is a bug?

Do we have a test to check this ?

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java [hunk 2]
```diff
@@ -572,24 +572,30 @@ protected boolean fillSnapshot(
...
-        snapshot.setExit(exitContext);
-      }
-      snapshot.addEvaluationErrors(exitStatus.getErrors());
+      fillEvaluationErrors(exitContext, snapshot, exitStatus);
       shouldCommit = true;
     }
     return shouldCommit;
   }
 
+  private static void fillEvaluationErrors(
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java:585]: suggest rename to popuplateSnapshotFromContextAndErrors

**Review by shatzi — COMMENTED** [review]
I think there is a bug here... please check that setEntry/setExit works

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java:589]: good catch, fixed

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java (hunk 2 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java:585]: renamed, but shorter

**Review by shatzi — APPROVED** [review]

**Pull request merged**